csmith-fuzzing/driver.py

```diff
@@ -35,31 +47,234 @@ def run_logged(cmd):
+        error_msg = match.group(1)
+        print("...searching for \"{}\".".format(error_msg))
+        return creduce_with_predicate_flags(
+            args,
+            failing_test_case,
+            "--bindgen-args '{}' --expect-compile-fail --rustc-grep '{}'".format(
+                args.bindgen_args,
+                re.escape(error_msg)
+            )
+        )
+
+    match = re.search(LAYOUT_TEST_FAILURE, result.stdout)
+    if match:
+        layout_failure = match.group(1)
+        struct_name = layout_failure[len("test bindgen_test_layout_"):layout_failure.rindex(" ... FAILED")]
+        print("...searching for \"{}\".".format(layout_failure))
+        return creduce_with_predicate_flags(
+            args,
+            failing_test_case,
+            "--bindgen-args '{}' --expect-layout-tests-fail --bindings-grep '{}' --layout-tests-grep '{}'".format(
+                args.bindgen_args,
+                re.escape(struct_name),
+                re.escape(layout_failure)
+            )
+        )
+
+    print("...nevermind, don't know how to `creduce` this bug. Skipping.")
+
+def creduce_with_predicate_flags(args, failing_test_case, predicate_flags):
+    predicate = """
+#!/usr/bin/env bash
+set -eu
+{} {} {}
+    """.format(
+        os.path.abspath(os.path.join(os.path.dirname(sys.argv[0]), "predicate.py")),
+        predicate_flags,
+        os.path.basename(failing_test_case)
+    )
+
+    print("...and reducing with this script:")
+    print()
+    print()
+    print(predicate)
+    print()
+    print()
+
+    predicate_path = failing_test_case + ".predicate.sh"
+    with open(predicate_path, "w") as p:
+        p.write(predicate)
+    os.chmod(predicate_path, 0o755)
+
+    creduce_command = ["creduce", "--n", str(os.cpu_count()), predicate_path, failing_test_case]
+    print("Running:", creduce_command)
+    result = run(creduce_command)
+    if result.returncode == 0:
+        print()
+        print()
+        print("`creduce` reduced the failing test case to:")
+        print()
+        print()
+        cat(failing_test_case)
+        print()
+        print()
+    else:
+        print()
+        print()
+        print("`creduce` failed!")
+        if not args.keep_going:
+            sys.exit(1)
+
+def print_issue_template(args, failing_test_case, predicate_command, result):
+    test_case_contents = None
+    with open(failing_test_case, "r") as f:
+        test_case_contents = f.read()
+
+    print("""
+
+! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! !
+! File this issue at https://github.com/rust-lang-nursery/rust-bindgen/issues/new !
+! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! !
+
+   --------------- 8< --------------- 8< --------------- 8< ---------------
+
+This bug was found with `csmith` and `driver.py`.
+
+### Input Header
+
+```c
+{}
+```
+
+### `bindgen` Invocation
+
+```
+$ {}
+```
+
+### Actual Results
+
+<details>
+
+```
+{}
```

Comment:
Yeah, this is true. `creduce` hides the output of the predicate script as it runs it (since it is running many in parallel), so the only way to get the output on the reduced test case would be to run `predicate.py` once again. I didn't think it was really worth the trouble...

Do you think it's important?

Reply:
Actually no, just nice to have!